source/detail/parsers.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024 xlnt-community
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+//
+// @license: http://www.opensource.org/licenses/mit-license.php
+// @author: see AUTHORS file
+
+#pragma once
+
+#include <string>
+#include <locale>
+#include <xlnt/xlnt_config.hpp>
+
+/// The following parsers offer the same convenience as C++'s parsers (std::stoll, std::stod, etc.) but do NOT require exception handling,
+/// resulting in improved performance and convenience to the programmer (exception handling is not required everywhere).
+namespace xlnt {
+namespace detail {
+
+/// ----- (SIGNED) INTEGER PARSING -----
+
+XLNT_API bool parse(const char *string, long long &result, char **end = nullptr, int base = 10);
```

Comment:
I know this is weird, but it is unfortunately necessary in order for our unit test project to be able to test these functions that are in another project.

Reply:
Ah ok. I understand now. Maybe we should use XLNT_API_INTERNAL to allow exposing internal APIs only if explicitly requested. But I'm fine to postpone it to later. I even think this macro does nothing on a non MSVC compiler at the moment.

Reply:
Correct, this macro is only necessary in order for MSVC to export the symbols. It does nothing on other compilers. (Note: nowadays, some compilers like GCC actually accept attribues like `__declspec(dllexport)` so that the code compiles, but these attributes are ignored.) This is necessary for the function to be exported correctly and to generate a `.lib` file that exports the function, so that the project consuming the library (in this case, the test project) links correctly. All other users of the exported functions want to import it, which is done by `__declspec(dllimport)`.

I did not think it through yet, but the only way I can imagine how `XLNT_API_INTERNAL` would work would be if we had a DLL for the XLNT library itself which should NOT be used for unit testing, and another DLL containing internal stuff that is only made for unit testing and would be used by the test project (instead of the normal DLL). This would increase compilation times, but would only do so when the unit tests are enabled in CMake. While this is not great, if we want to test more internal functions in the future, this will be necessary to avoid exporting many internal functions. However, I would prefer to have another PR for this (where we can discuss further how this should work), and simply export these few functions for now :slightly_smiling_face:

Reply:
Ok

Reply:
Note that I added in my latest PR an `XLNT_API_INTERNAL` define, especially for this usecase. Once it is merged, you can use it too.